src/Purebred/Types/AddressBook.hs

```diff
@@ -0,0 +1,68 @@
+-- This file is part of purebred
+-- Copyright (C) 2022 Róman Joost
+--
+-- purebred is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU Affero General Public License as published by
+-- the Free Software Foundation, either version 3 of the License, or
+-- (at your option) any later version.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+{- |
+
+Types for handling the retrieval of addresses
+
+-}
+module Purebred.Types.AddressBook
+  (
+    AddressRetriever(..)
+  , AddressRetrieverConfig(..)
+  , runAddressRetriever
+  , MuttAlias(..)
+  , muttAliasNick
+  , muttAliasAddress
+  ) where
+
+import Control.Lens (Lens', lens)
+import qualified Data.Text as T
+import GHC.Generics (Generic)
+import Control.DeepSeq (NFData(rnf))
+
+import Data.IMF (Address)
+
+data AddressRetrieverConfig = MuttAliasConfig FilePath | LDAPConfig String
+  deriving (Generic, NFData)
+
+newtype AddressRetriever = AddressRetriever
+  { _runAddressRetriever :: T.Text -> IO [Address]
+  }
+  deriving (Generic)
+
+instance NFData AddressRetriever where
+  rnf (AddressRetriever retrieve) = AddressRetriever retrieve `seq` ()
+
+runAddressRetriever :: Lens' AddressRetriever (T.Text -> IO [Address])
+runAddressRetriever = lens _runAddressRetriever (\r x -> r { _runAddressRetriever = x })
+
+data MuttAlias = MuttAlias
+  { _muttAliasNick :: T.Text,
+    _muttAliasAddress :: Address
+  } deriving (Show, Eq)
+
+muttAliasNick :: Lens' MuttAlias T.Text
+muttAliasNick = lens _muttAliasNick (\m x -> m { _muttAliasNick = x })
+
+muttAliasAddress :: Lens' MuttAlias Address
+muttAliasAddress = lens _muttAliasAddress (\m x -> m { _muttAliasAddress = x })
```

Comment:
We should move these definitions to `Purebred.Storage.AddressBook.MuttAliasFile`.

Reply:
moved